morebits.js

```diff
@@ -4566,215 +4577,1725 @@ Morebits.wiki.page = function(pageName, status) {
+	}
+	// Normalize IPv6
+	userName = Morebits.ip.sanitizeIPv6(userTitle.getMainText());
+
+	if (!currentAction) {
+		currentAction = 'Querying user "' + userName + '"';
+	}
 
 	/**
-	 * Displays the preview box, and begins an asynchronous attempt
-	 * to render the specified wikitext.
+	 * Private context variable not visible to the outside, thus all the
+	 * data here must be accessed via getter and setter functions.
 	 *
-	 * @param {string} wikitext - Wikitext to render; most things should work, including `subst:` and `~~~~`.
-	 * @param {string} [pageTitle] - Optional parameter for the page this should be rendered as being on, if omitted it is taken as the current page.
-	 * @param {string} [sectionTitle] - If provided, render the text as a new section using this as the title.
-	 * @returns {jQuery.promise}
+	 * @private
 	 */
-	this.beginRender = function(wikitext, pageTitle, sectionTitle) {
-		$(previewbox).show();
+	var ctx = {
+		userName: userName,
+		userID: null,
+		editCount: null,
+		registration: null,
+		exists: false,
+		hidden: false,
+		loadTime: null,
+		isIP: mw.util.isIPAddress(userName, true),
+		isIPRange: Morebits.ip.isRange(userName),
 
-		var statusspan = document.createElement('span');
-		previewbox.appendChild(statusspan);
-		Morebits.status.init(statusspan);
+		reason: null, // Will default to current reason if reblocking
+		changeTags: null,
+		watchuser: false,
+		watchlistExpiry: null,
+		expiry: null, // Will default to current expiry if reblocking
+		callbackParameters: null,
+		statusElement: currentAction instanceof Morebits.status ? currentAction : new Morebits.status(currentAction),
+
+		// block paramters
+		hasBlockLog: null,
+		lastBlockLogEntry: null,
+		blockInfo: null, // If blocked, an object full of block parameters
+		blockedRange: null,
+		isBlocked: null,
+		isRangeBlocked: null,
+		reblock: false,
+		useOriginalBlockParams: true,
+
+		// null before loading, array after
+		groups: null,
+		autoGroups: null,
+		userRights: null,
+		// Response is array of objects with group: name and expiry: time,
+		// but we force it into an object with groupname: expiration
+		grantedGroups: null,
+
+		/* Block */
+		// If the user is directly blocked, these defaults will be
+		// overridden with values from the active block unless
+		// useOriginalBlockParams is set
+		allowusertalk: true,
+		anononly: false,
+		autoblock: true,
+		nocreate: true,
+		noemail: false,
+		hidename: false,
+		partial: false,
+		namespacerestrictions: null,
+		pagerestrictions: null,
+
+		/* Change usergroups */
+		addGroups: null,
+		removeGroups: null,
+
+		/* Notify */
+		// talkLinks, talkTemplates, and notifySkipTemplates end up as arrays
+		talkTitle: userTitle.getTalkPage().toText(),
+		talkText: null,
+		talkExists: null,
+		talkTimestamp: null,
+		talkLastEditor: null,
+		talkTemplates: null,
+		talkLinks: null,
+		message: null,
+		sectionTitle: null,
+		notifyBots: false,
+		notifyIndef: false,
+		notifySelf: false,
```

Comment:
Pointless?  Unused atm because it's nicer to quit earlier but should this even be an option, or just always a given?